mm2src/mm2_io/src/fs.rs

```diff
@@ -192,19 +192,53 @@ where
     json::from_slice(&content).map_to_mm(FsJsonError::Deserializing)
 }
 
-/// Read the `dir_path` entries trying to deserialize each as the `T` type.
+async fn filter_files_by_extension(dir_path: &Path, extension: &str) -> IoResult<Vec<PathBuf>> {
+    let ext = Some(OsStr::new(extension).to_ascii_lowercase());
+    let entries = read_dir_async(dir_path)
+        .await?
+        .into_iter()
+        .filter(|path| path.extension().map(|ext| ext.to_ascii_lowercase()) == ext)
+        .collect();
+    Ok(entries)
+}
+
+/// Helper function to extract file names or stems based on the provided extraction function.
+fn extract_file_identifiers<F>(entries: Vec<PathBuf>, extractor: F) -> Vec<String>
+where
+    F: Fn(&Path) -> Option<&OsStr>,
+{
+    entries
+        .into_iter()
+        .filter_map(|path| extractor(&path).and_then(OsStr::to_str).map(ToOwned::to_owned))
+        .collect()
+}
```

Comment:
we can delegate collection to caller 
```suggestion
fn extract_file_identifiers<F>(entries: Vec<PathBuf>, extractor: F) -> impl Iterator
where
    F: Fn(&Path) -> Option<&OsStr>,
{
    entries
        .into_iter()
        .filter_map(|path| extractor(&path).and_then(OsStr::to_str).map(ToOwned::to_owned))
}
```

Reply:
fixed here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2202/commits/de645bdd394c6090d84a6812f4580ff51f0df91d